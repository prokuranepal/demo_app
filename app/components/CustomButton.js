import React from 'react';
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native';

import PropTypes from 'prop-types';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors';

const CustomButton = (props) => (
  <View style={props.container_style}>
    <TouchableOpacity
      style={[styles.saveStyle, props.buttonStyle]}
      onPress={props.pressHandler}
      data-test="touchableComp">
      <Text style={{ ...styles.text,color: props.color}} data-test="textData">{props.title}</Text>
    </TouchableOpacity>
  </View>
)

const styles = StyleSheet.create({
  text: {
    fontSize: fontSize.buttonText
  },
  saveStyle: {
    borderRadius: 40,
    paddingHorizontal: 60,
    paddingVertical: 10,
    backgroundColor: colors.primaryButton
  }
})

CustomButton.propTypes = {
  buttonStyle: PropTypes.object,
  containerStyle: PropTypes.object,
  color: PropTypes.string,
  title: PropTypes.string,
  pressHandler: PropTypes.func
}


export default CustomButton;