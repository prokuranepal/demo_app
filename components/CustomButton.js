import React from 'react';
import { View, Button, TouchableOpacity, Text,StyleSheet } from 'react-native';
const customButton =(props)=>(
    <View style={props.container_style}>
        <TouchableOpacity
          style={[styles.saveStyle,props.buttonStyle]}
          onPress={ props.pressHandler }>
            <Text style={{color:props.color, fontSize:17}}> {props.title} </Text>
      </TouchableOpacity>
</View>
)

const styles = StyleSheet.create({
saveStyle:{
    borderRadius:40,
    paddingHorizontal:60,
    paddingVertical:10,
    backgroundColor:"#4fbed2"
  }
})

export default customButton;