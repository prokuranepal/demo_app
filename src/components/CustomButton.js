import React from 'react';
import { View, Button, TouchableOpacity, Text, StyleSheet } from 'react-native';
import {colors} from '../ThemeColors/themeColors';
const customButton = (props) => (
  <View style={props.container_style}>
    <TouchableOpacity
      style={[styles.saveStyle, props.buttonStyle]}
      onPress={props.pressHandler} data-test="touchableComp">
      <Text style={{ color: props.color, fontSize: 17 }} data-test="textData">{props.title}</Text>
    </TouchableOpacity>
  </View>
)

const styles = StyleSheet.create({
  saveStyle: {
    borderRadius: 40,
    paddingHorizontal: 60,
    paddingVertical: 10,
    backgroundColor: colors.primaryButton
  }
})

export default customButton;