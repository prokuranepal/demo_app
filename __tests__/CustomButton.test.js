import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import {View, Text, TouchableOpacity} from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import CustomButton from '../src/components/CustomButton';
import {findByTestAttr} from '../src/utils/testFunction';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()
const dummy_data = {
    title:"Title Test",
    pressHandler:function_click
}


describe('<CustomButton />', () => {

    it('should render without issues', () => {
        const component = shallow(<CustomButton />);
        expect(component.length).toBe(1);
        expect(component).toMatchSnapshot();
      });

    it("Text Component and Title Prop test in TestInput", () =>
      {
          const wrapper = shallow(<CustomButton  {...dummy_data} />)
          let textData = findByTestAttr(wrapper, "textData")
          expect(textData.length).toBe(1);
          expect(textData.props().children).toEqual("Title Test")
          expect(wrapper.find(View)).toHaveLength(1)          
          expect(wrapper.find(Text)).toHaveLength(1)
          expect(wrapper.find(TouchableOpacity)).toHaveLength(1)          
      
     });
    it("Touch Component and Event test", () =>
     {
        const wrapper = shallow(<CustomButton  {...dummy_data} />)
        let touchableComp = findByTestAttr(wrapper, "touchableComp")
        expect(touchableComp).toHaveLength(1)
        touchableComp.props().onPress()
        wrapper.update()
        expect(function_click).toHaveBeenCalledTimes(1)
     })


})