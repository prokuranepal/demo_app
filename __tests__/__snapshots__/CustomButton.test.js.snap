// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> should render without issues 1`] = `ShallowWrapper {}`;
